packages/phone-number-privacy/combiner/src/common/sign.ts

```diff
@@ -7,16 +7,27 @@ import {
 } from '@celo/phone-number-privacy-common'
 import { Response as FetchResponse } from 'node-fetch'
 import { OdisConfig } from '../config'
+import { DomainThresholdStateService } from '../domain/services/thresholdState'
+import { CombinerThresholdStateService } from '../pnp/services/thresholdState'
 import { CombineAction } from './combine'
 import { CryptoSession } from './crypto-session'
 import { IO } from './io'
 
 // prettier-ignore
 export type OdisSignatureRequest = SignMessageRequest | DomainRestrictedSignatureRequest
+export type ThresholdStateService<R> = R extends SignMessageRequest
```

Comment:
Nit: This might be a bit redundant but it would be clearer and more consistent with code elsewhere if this were ThresholdStateService<R extends OdisSignatureRequest>.